syntax = "proto2";
import "graph_structures.proto";
import "google/protobuf/timestamp.proto";
package ujt_protos;

service ReportingService {
    // for generating graph
    rpc GetNodes(GetNodesRequest) returns (GetNodesResponse) {}
    // for updating statuses
    rpc GetSLIs(GetSLIsRequest) returns (GetSLIsResponse) {}
    // for user journeys
    rpc GetClients(GetClientsRequest) returns (GetClientsResponse) {}

}

message GetNodesRequest {
    // We can add filtering fields later.
    // Not sure if it's necessary/useful to request a subset of nodes.
    // I imagine nodes should be requested once at startup (and perhaps force-refreshed), 
    // which should both request all nodes.
}

message GetNodesResponse {
    repeated Node nodes = 1;
}

message GetSLIsRequest {
    repeated string node_names = 1;
    repeated SLIType sli_types = 2;
    optional google.protobuf.Timestamp start_time = 3;
    optional google.protobuf.Timestamp end_time = 4;
}

message GetSLIsResponse {
    repeated SLI slis = 1;
}

message GetClientsRequest {
    // ...
}

message GetClientsResponse {
    repeated Client clients = 1;
}
